File: frontend/src/types/emprestimo.ts
import type { Usuario } from './usuario';
import type { Livro } from './livro';

export type Emprestimo = {
    id: number;
    usuario: Usuario;
    livro: Livro;
    dataEmprestimo: string;
    dataDevolucao?: string;
    status: string;
};

export type CreateEmprestimoData = {
    usuarioId: number;
    livroId: number;
    dataEmprestimo: string;
};

File: backend/src/main/java/com/elotech/biblioteca/model/Emprestimo.java
package com.elotech.biblioteca.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDate;

@Entity
@Table(name = "emprestimos")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Emprestimo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "usuario_id", nullable = false)
    private Usuario usuario;

    @ManyToOne
    @JoinColumn(name = "livro_id", nullable = false)
    @NotNull(message = "O livro é obrigatório.")
    private Livro livro;

    @Column(name = "data_emprestimo", nullable = false)
    @NotNull(message = "A data de empréstimo é obrigatória.")
    private LocalDate dataEmprestimo;

    @Column(name = "data_devolucao")
    private LocalDate dataDevolucao;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private StatusEmprestimo status;
}